crates/contracts/uuid-tld/src/lib.rs

```diff
@@ -105,28 +71,103 @@ impl UuidTld {
         hasher.write(tx_ctx.block_hash.0.as_bytes());
         hasher.write_u128(tx_ctx.timestamp);
         let mut hasher_rng = hasher.into_rng();
-        let id = uuid::Builder::from_random_bytes(hasher_rng.random()).into_uuid();
+        let id = uuid::Builder::from_random_bytes(hasher_rng.random())
+            .into_uuid()
+            .as_u128();
 
         // _really_ shouldn't happen but let's handle it regardless.
-        if !self.registered_contracts.insert(id.as_u128()) {
+        if self.registered_contracts.contains_key(&id) {
             return Err("Contract already registered".to_string());
         }
 
-        info!("Registering new contract with UUID {}", id);
-
+        self.registered_contracts
+            .insert(id, contract_input.identity.0.clone());
+        let uuid = Uuid::from_u128(id);
+        info!("Claimed new contract with UUID {} ({})", uuid, id);
         Ok(id)
     }
+
+    pub fn register_contract(&mut self, uuid: u128, identity: String) -> Result<(), String> {
+        match self.registered_contracts.get(&uuid) {
+            Some(claimer) if claimer == &identity => {
+                info!(
+                    "Registering contract with UUID {} ({})",
+                    Uuid::from_u128(uuid),
+                    uuid
+                );
+                Ok(())
+            }
+            Some(_) => Err("UUID claimed by different identity".to_string()),
+            None => Err("UUID not claimed".to_string()),
+        }
+    }
+}
+
+impl HyleContract for UuidTld {
+    fn execute(&mut self, contract_input: &ContractInput) -> RunResult {
```

Comment:
I would prefer keep that entrypoint on top of file like before, this way all our contrats "look the same" when someone open the file